specs/interop/token-bridging.md

```diff
@@ -0,0 +1,97 @@
+# Token Bridging
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Interface](#interface)
+- [Implementation](#implementation)
+- [L1 Native Tokens](#l1-native-tokens)
+- [Cross Chain `transferFrom`](#cross-chain-transferfrom)
+- [Factory](#factory)
+- [Upgrading Existing `OptimismMintableERC20Token`s](#upgrading-existing-optimismmintableerc20tokens)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Unified ERC20 token bridging built on top of the messaging protocol
+can enable tokens to be fungible and movable across the superchain.
+An issue with bridging is that if the security model is not standardized,
+then the bridged assets are not fungible with each other depending on
+which bridge solution was used to facilitate the transfer between
+domains.
+
+## Interface
+
+The `ISuperchainERC20` interface creates a standard way to send tokens
+between domains. In contrast to [xERC20][xerc20] where the user calls
+a bridge that has `mint` and `burn` privileges on the cross chain enabled
+tokens, the user instead calls methods on the token contract directly.
+
+[xerc20]: https://www.xerc20.com/
+
+```solidity
+interface ISuperchainERC20 {
+  function bridgeERC20(uint256 amount, uint256 chainId) external;
+  function bridgeERC20To(address to, uint256 amount, uint256 chainId) external;
+  function finalizeBridgeERC20(address to, uint256 amount) external;
+}
+```
+
+## Implementation
+
+An example implementation that depends on deterministic deployments across chains
+for security is provided. This construction builds on top of the [L2ToL2CrossDomainMessenger][l2-to-l2]
+for both replay protection and domain binding.
+
+[l2-to-l2]: ./predeploys.md#l2tol2crossdomainmessenger
+
+```solidity
+contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
+  constructor(string memory _name, string memory _symbol, uint256 _decimals) ERC20(_name, _symbol, _decimals) {}
+
+  function bridgeERC20(uint256 _amount, uint256 _chainId) external {
+    bridgeERC20To(msg.sender, _amount, _chainId);
+  }
```

Comment:
Not sure if we care about this for the reference implementation, but assuming we want this for aliasing reasons
```suggestion
  function bridgeERC20(uint256 _amount, uint256 _chainId) external {
    require(msg.sender == tx.origin, "OptimismSuperchainERC20: function can only be called from an EOA");
    bridgeERC20To(msg.sender, _amount, _chainId);
  }
```

Reply:
Ah yes good call, then its similar to the existing bridge implementation

Reply:
does this still made sense under eip 3074?

Reply:
This check exists to prevent a footgun with aliasing between L1 and L2 where you send to the unaliased account or you send to an account that you don't control since there is no guarantee you control the same smart contract on two different domains.

Since there is no aliasing between domains, its possible to argue that this check isn't needed. I would want to see an example of a 4337 bundler - i believe the `msg.sender` would be the smart contract wallet. Ideally a user controls the same account between every domain but its not guaranteed. So it would be an implicit footgun if we allowed for it